internal/procedures/server/start.go

```diff
@@ -14,9 +14,13 @@ import (
 	"github.com/edulinq/autograder/internal/util"
 )
 
-func Start(initiator common.ServerInitiator) (err error) {
-	defer server.StopServer()
+type ProcedureServer struct {
+	apiServer *server.APIServer
+}
+
+var serverInstance *ProcedureServer
```

Comment:
When defining variables like this, I prefer explicitly setting their value (`= nil`).
It does not change the program, but makes it more clear.